### PR TITLE
Use `in_array` instead of `array_search` + comparison

### PR DESCRIPTION
**Pull request by carusogabriel**
### Description (*)
This has some benefits, such as reducing the tasks, as well as improving the readability of the code.

### Fixed Issues (if relevant)
None

### Manual testing scenarios (*)
Not necessary

### Questions or comments
-

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by osrecio** [comment]
Hi @carusogabriel . Thanks for your PR. 

Can you check the [tests details](https://testing-service.magento-community.engineering/reports/magento/magento2/pull/24886/4b797d19453cab9949081026ac6fb49b/Statics/allure-report-ce/index.html#suites/51d3ab4d5d77a0bc6a7ea344e9e7e6be/715df97677547e19/) and solve the errors.

**Comment by carusogabriel** [comment]
Hello @osrecio. I thought the failings were false-positives [in my PRs](https://github.com/magento/magento2/pulls/carusogabriel). I'll fix them, thanks 👍

File: app/code/Magento/Analytics/Model/Cryptographer.php [hunk 1]
```diff
@@ -135,6 +135,6 @@ private function validateCipherMethod($cipherMethod)
         );
         $cipherMethod = strtolower($cipherMethod);
 
-        return (false !== array_search($cipherMethod, $methods));
+        return in_array($cipherMethod, $methods);
```
Comment (ihor-sviziev) [inline app/code/Magento/Analytics/Model/Cryptographer.php:138]: I would like to replace both array_search & in_array to array_flip + isset.
Reason: https://github.com/magento/magento-coding-standard/issues/143

@carusogabriel @osrecio what do you think about it?

Comment (osrecio) [reply app/code/Magento/Analytics/Model/Cryptographer.php:138]: Followin the thread I think is better to usee `array_flip` + `isset`.

If @carusogabriel agree with us, he can perform the changes and for me will be ready to be merged :)

Comment (carusogabriel) [reply app/code/Magento/Analytics/Model/Cryptographer.php:138]: @osrecio I'll do it :)

**Pull request closed without merging**